packages/canisters/src/ic_af/src/lib.rs

```diff
@@ -67,17 +27,47 @@ fn sha256(input: &[u8]) -> [u8; 32] {
 /// verifying the signatures for data-package
 /// Sample file : ../../../../fixtures/data-package.json
 #[update]
-fn verify_data_proof(data_package : String) -> (String, String) {
-    let data_package : DataPackage = serde_json::from_str(data_package.as_str()).unwrap();
-    // todo!();
-    (String::new(), String::new())
+async fn verify_data_proof(redstone_data : String) -> (Vec<DataPackage>, String) {
```

Comment:
This data proof should technically be for any data received over Streamr.
Redstone data packages are transported over Streamr.
This means, 
1. verify that the Streamr message is published from the authorised Redstone publisher first.
2. verify the redstone data package

Reply:
I modify `verify_data_proof` function and create a new enum type `StreamData` for verifying various types of stream data. I also updated to check if the streamr id is valid Redstone publisher.

Reply:
This is not enough.

The data may, or may NOT, be Redstone data.
If it is NOT, then it's likely generic data from Streamr.

As I previously mentioned the first verification step is to verify the Streamr metadata.

Then the second step may be that the Streamr data came from RedStone, but it might also NOT come from RedStone.

As you can see https://github.com/usherlabs/ccamp/blob/130a447bb4c816c7fe99288ec53083d329ee6976/packages/canisters/src/protocol_data_collection/src/remittance.rs#L21 - we've already included ways to verify Streamr data.

I'll need you to use this logic as implementation here.